### ISSUE TITLE
[1.21.x][Neoforge] Mod causing game to not tick and requires me to force close the game

### ISSUE THREAD

**Issue by ffuentesm** [Ramixin/VisibleTraders#14]
Bug Description
After trying to /reload my world things are weird. I cant use item, I can only run and jump, and the running does not stop, and new chunks will not load. Looking at the log this message is where it starts
[ModernFix integrated server watchdog/ERROR] [org.embeddedt.modernfix.world.IntegratedWatchdog/]: A single server tick has taken 40001, more than 40000 milliseconds

Log File
[2024-09-12-7.log](https://github.com/user-attachments/files/16988328/2024-09-12-7.log)

Looking at the log, the visibletraders mod appears to be the issue; it is trying to load chunks in the middle of a structure trying to spawn with villagers.

**Comment by Ramixin** [comment]
Hello @ffuentesm . For future cases, please upload logs to https://mclo.gs since it makes the process more streamlined. It appears that Visible Traders is trying to generate a map for some sort of structure, but that structure could not be located for some reason. Please run an instance with only Visible Traders loaded and try to reproduce the issue. If it does, please then also share the refined log

**Comment by TelepathicGrunt** [comment]
It is this line:
https://github.com/Ramixin/VisibleTraders/blob/64571528130e61888c8c01347f5995abfad98b83/src/main/java/net/ramixin/visibletraders/mixins/VillagerMixin.java#L107

Someone reported the issue to my mod thinking it was me. It was not.
https://github.com/TelepathicGrunt/RepurposedStructures/issues/350

Look at Worker-Main-5 in that person's thread dump. The VisibleTraders mod is causing a villager spawned during structure worldgen to run the code to generate its trades. One of the trade is a structure map. And this map item now gets loaded and needs t find the structure it is looking for. However, since this is in worldgen within a chunk already being generated, it causes the chunk to wait on itself and deadlock.

**Comment by Ramixin** [comment]
@ffuentesm and @willyougooutwithme , I have attempted to fix the issue, but I am not sure if it is patched since I was unable to recreate the issue myself. Before releasing a new version, it would be very helpful if you could test a beta build with the supposed fix incase it doesn't work. (You will need to change the file extension from .zip to .jar since GitHub won't let me upload jars)
[VisibleTraders-0.0.6-BETA.zip](https://github.com/user-attachments/files/17177105/VisibleTraders-0.0.6-BETA.zip)
[VirusTotal Report](https://www.virustotal.com/gui/file/cd547c9be9a376123b055fcbac0c80e69db4ce9bf1725e30e9679cc906e04c6c?nocache=1)

**Comment by ffuentesm** [comment]
Hey, I tested  but the zip provided does not work
![image](https://github.com/user-attachments/assets/095e493e-a164-4099-b32c-3fe28b9f1f3c)

**Comment by Ramixin** [comment]
The change was merged with V:0.0.6 for Minecraft 1.20.5 through 1.21.x. Since I still haven't been able to verify everything is working as intended, I will keep this issue open

**Comment by Keaton42** [comment]
I'm having the same issue with Fabric 1.21.1 unfortunately

**Comment by Ramixin** [comment]
@Keaton42 , Can you share a log through https://mclo.gs for me to check?

**Comment by Keaton42** [comment]
Yeah! I’ll do that in the afternoon and see what’s causing the trouble. I wonder if it’s compatibility but not sure. Also great mod, this idea is fun and useful and should already be in Minecraft!
